### DISCUSSION TITLE
Is this an error in readme

### DISCUSSION THREAD

**Discussion by iacore** [vnmakarov/mir#326] [Q&A] [answered]
I read this in `readme.md`

> `sieve:    func i32, i32:N`
> ...
> func describes signature of the function (taking 32-bit signed integer argument and returning 32-bit signed integer value) and function argument N which will be local variable of 64-bit signed integer type

I think it should be
> function argument N which will be local variable of 32-bit signed integer type

**Comment by vnmakarov** [comment, marked as answer]
This is not an error.  All integer variables are always 64-bit integers.  Therefore you can not use smaller integer types in `local`.
Argument and return types are for function interface.  Basically value of an argument described as i32 will be signed extended and stored in local 64-bit variable `N`. 

May be such type system is not obvious but it is significantly simplifies the implementation without big performance impact.  The impact is due to additional 32-bit to 64-bit extensions but many such extensions will be removed by the generator optimizations.
